internal/db/collection_define.go

```diff
@@ -25,113 +25,126 @@ import (
 	"github.com/sourcenetwork/defradb/internal/db/description"
 )
 
-func (db *db) createCollection(
+func (db *db) createCollections(
 	ctx context.Context,
-	def client.CollectionDefinition,
 	newDefinitions []client.CollectionDefinition,
-) (client.Collection, error) {
-	schema := def.Schema
-	desc := def.Description
-	txn := mustGetContextTxn(ctx)
-
-	if desc.Name.HasValue() {
-		exists, err := description.HasCollectionByName(ctx, txn, desc.Name.Value())
-		if err != nil {
-			return nil, err
-		}
-		if exists {
-			return nil, ErrCollectionAlreadyExists
-		}
-	}
+) ([]client.CollectionDefinition, error) {
+	returnDescriptions := make([]client.CollectionDefinition, len(newDefinitions))
 
 	existingDefinitions, err := db.getAllActiveDefinitions(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	schemaByName := map[string]client.SchemaDescription{}
-	for _, existingDefinition := range existingDefinitions {
-		schemaByName[existingDefinition.Schema.Name] = existingDefinition.Schema
-	}
-	for _, newDefinition := range newDefinitions {
-		schemaByName[newDefinition.Schema.Name] = newDefinition.Schema
-	}
-
-	_, err = validateUpdateSchemaFields(schemaByName, client.SchemaDescription{}, schema)
-	if err != nil {
-		return nil, err
-	}
-
-	definitionsByName := map[string]client.CollectionDefinition{}
-	for _, existingDefinition := range existingDefinitions {
-		definitionsByName[existingDefinition.GetName()] = existingDefinition
-	}
-	for _, newDefinition := range newDefinitions {
-		definitionsByName[newDefinition.GetName()] = newDefinition
-	}
-	err = db.validateNewCollection(def, definitionsByName)
-	if err != nil {
-		return nil, err
-	}
+	for i, def := range newDefinitions {
+		schema := def.Schema
```

Comment:
todo: Since schema is not used before being reassigned, please use `def.Schema` directly in `CreateSchemaVersion` bellow and instanciate `schema` at that time.

Reply:
Nice spot, removing.

- [ ] sort out `schema` vars